components/sup/src/main.rs

```diff
@@ -1249,6 +1246,105 @@ sys_ip_address = "7.8.9.0"
+            let mut config_file = File::create(&config_path).unwrap();
+            write!(config_file, "{}", config_contents).expect("to write config file contents");
+
+            let args = format!("hab-sup run --config-files {}", config_path_str);
+
+            let gossip_peers = vec!["1.1.1.1:9638".parse().unwrap(),
+                                    format!("127.0.0.1:{}", GossipListenAddr::DEFAULT_PORT).parse()
+                                                                                         .unwrap()];
+
+            let config = config_from_cmd_str(&args);
+            assert_eq!(ManagerConfig { auto_update: true,
+                                       auto_update_period: Duration::from_secs(3600),
+                                       service_update_period: Duration::from_secs(1_000),
+                                       service_restart_config: ServiceRestartConfig::default(),
+                                       custom_state_path: None,
+                                       key_cache: KeyCache::new(temp_dir_str),
+                                       update_url: String::from("https://bldr.habitat.sh"),
+                                       update_channel: ChannelIdent::default(),
+                                       gossip_listen:
+                                           GossipListenAddr::from_str("1.2.3.4:4321").unwrap(),
+                                       ctl_listen:
+                                           ListenCtlAddr::from_str("7.8.9.1:12").unwrap(),
+                                       ctl_server_certificates: None,
+                                       ctl_server_key: None,
+                                       ctl_client_ca_certificates: None,
+                                       http_listen:
+                                           HttpListenAddr::from_str("5.5.5.5:11111").unwrap(),
+                                       http_disable: true,
+                                       gossip_peers,
+                                       gossip_permanent: true,
+                                       ring_key: Some(ring_key),
+                                       organization: Some(String::from("MY_ORG")),
+                                       watch_peer_file: None,
+                                       tls_config: Some(TLSConfig { cert_path,
+                                                                    key_path,
+                                                                    ca_cert_path:
+                                                                        Some(ca_cert_path) }),
+                                       feature_flags: FeatureFlag::empty(),
+                                       event_stream_config: None,
+                                       keep_latest_packages: Some(5),
+                                       sys_ip: "7.8.9.0".parse().unwrap() },
+                       config);
+        }
+
```

Comment:
I think you are really only interested in testing the peer values and not all of the other values. We already have a test that tests the other stuff and this test can then be MUCH smaller because you won't need all the cert key stuff or alot of other ceremony.

Reply:
Have modified the test case to specifically target at peer values.
Please refer  commit 8930f52d98b666253fb494c288a5c2cdb3bf686c